agent/consul/stream/subscription.go

```diff
@@ -68,16 +73,16 @@ func newSubscription(req SubscribeRequest, item *bufferItem, unsub func()) *Subs
 // Next returns the next Event to deliver. It must only be called from a
 // single goroutine concurrently as it mutates the Subscription.
 func (s *Subscription) Next(ctx context.Context) (Event, error) {
-	if atomic.LoadUint32(&s.state) == subscriptionStateClosed {
-		return Event{}, ErrSubscriptionClosed
-	}
-
 	for {
+		if err := s.stateIsOpen(); err != nil {
```

Comment:
This feels like a nit, but at a glance this err check had me scratching my head. Is the `if err := xIsOpen(); err != nil { return ... }` form for translating error cases a go idiom? I expected it to contain a bool. It makes perfect sense after looking at the impl. but threw me a bit before I read the fn. Maybe a quick doccomment here could help? e.g. `// Check for closed states and error out`. To be clear, the if - err assignment wasn't confusing so much as the fn name `s.stateIsOpen()` and switch inside of it. I would expect it to be named something like `s.extractError()` instead?

Reply:
This is a good call out, generally any `IsX` or `HasX` function should return a bool. I agree the function name is awkward. A better name would be something like `checkSubOpen`,  `ensureSubOpen`, `requireSubOpen`. I think those should imply an error return.

Any preference for those or something along those lines? I can try out one to see how it looks.

Reply:
I amended this commit with the renamed `requireStateOpen` and fixed some typos in docs. Also removed another var from `forceClosed` to `closed` since it is used in both cases now.